src/rpc/protocol.ts

```diff
@@ -0,0 +1,105 @@
+import { MuSchema } from '../schema/schema';
+import { MuUnion, MuVarint, MuStruct, MuUTF8 } from '../schema';
+
+export type MuRPCTableEntry<
+    ArgsSchema extends MuSchema<any>,
+    ReturnSchema extends MuSchema<any>> = {
+    arg:ArgsSchema;
+    ret:ReturnSchema;
+};
+
+export type MuRPCTable = {
+    [method:string]:MuRPCTableEntry<any, any>;
+};
+
+export type MuRPCProtocol<RPCTable extends MuRPCTable> = {
+    name:string;
+    methods:RPCTable;
```

Comment:
how about `procedures` or `procs`? considering not to be confused with "methods" in oop and the module is about rpc